core/src/helpers/formatToAutomaticDecimalPoints.ts

```diff
@@ -58,6 +58,10 @@ function limitedValue(value: number | BigNumber): number | BigNumber {
     return value;
 }
 
+export function formatWithThousandSeparators(value: string): string {
+    return value.replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ',');
```

Comment:
As suggested in the call, why not use `intl` for that? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat and make it country-specific out of the box?

Reply:
See PR description: 

> I decided to use a custom regex expression over native JavaScript number functions, because the native functions only support the number type. As we do a lot of other custom formatting, the end result is always a string, resulting in them being useless. In addition to this formatting the number before we add our automatic decimal place messes up the end results, so I thought it was easier to apply the thousand commas last with regex rather then before. It also helps with the fact that we use numbers, BigNumbers and strings in some places, allowing me to simplify the amount of types the function needs to support.

Reply:
Nit: I would add a _shorter_ comment above the line with regexp explaining this decision